### PR TITLE
Use default nextjs error page

### PR DESCRIPTION
**Pull request by rocketnova**
## Changes

- Display default next.js error page if an error is encountered when querying the API gateway

## Context

For #242, I want to be able to throw errors and have them caught & display a 500 error (we may want to display 404 in some cases, but I'm not sure which ones yet).

I attempted to move the [`getServerSideProps()`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) logic out of index.tsx and into main.tsx. However, I discovered that you can only use it in [Page level](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) components. In light of this, I have [changed my mind](https://github.com/cagov/ui-claim-tracker/pull/256#discussion_r658328707) and think that we should collapse main.tsx directly into index.tsx (not included in this PR). Let me know if this sounds good and I can make a ticket for it/feel free to make a ticket for it.

Note that we have a separate ticket (#123) for customizing the default next.js error pages.

## Test

`yarn storybook`

### CODE REVIEW

**Comment by lomky** [comment]
> I attempted to move the getServerSideProps() logic out of index.tsx and into main.tsx. However, I discovered that you can only use it in Page level components. In light of this, I have changed my mind and think that we should collapse main.tsx directly into index.tsx (not included in this PR). Let me know if this sounds good and I can make a ticket for it/feel free to make a ticket for it.

Sounds right to me. File the issue and let me know if you'd like to to pick it up so you can stay focused!

File: pages/index.tsx [hunk 1]
```diff
@@ -55,17 +68,28 @@ export const getServerSideProps: GetServerSideProps = async ({ req, locale }) =>
...
-  const claimData = await queryApiGateway(req)
+  let errorCode: number | null = null
+  let claimData: Claim | null = null
+  let scenarioContent: ScenarioContent | null = null
 
-  // Run business logic to get content for the current scenario.
-  const scenarioContent = getScenarioContent(claimData)
+  try {
+    // Make the API request and return the data.
+    claimData = await queryApiGateway(req)
```
Comment (lomky) [inline pages/index.tsx:77]: Nit: Any reason not to declare `claimData` here instead of the outer scope?

Comment (rocketnova) [reply pages/index.tsx]: Good catch! Pushed a commit to correct this.

**Comment by rocketnova** [comment]
> > I attempted to move the getServerSideProps() logic out of index.tsx and into main.tsx. However, I discovered that you can only use it in Page level components. In light of this, I have changed my mind and think that we should collapse main.tsx directly into index.tsx (not included in this PR). Let me know if this sounds good and I can make a ticket for it/feel free to make a ticket for it.
> 
> Sounds right to me. File the issue and let me know if you'd like to to pick it up so you can stay focused!

Opened #265. Would you mind picking it up?

**Review by lomky — APPROVED** [review]

**Pull request merged**